packages/did-provider-ebsi/src/EbsiDidProvider.ts

```diff
@@ -120,11 +128,69 @@ export class EbsiDidProvider extends AbstractIdentifierProvider {
     throw Error(`Not (yet) implemented for the EBSI did provider`)
   }
 
+  private assertedKey = (args: { key?: IKeyOpts; type: EbsiKeyType; kms?: string }): MinimalImportableKey => {
+    const { key, type, kms } = args
+    const minimalImportableKey: Partial<MinimalImportableKey> = { ...key } ?? {}
+    minimalImportableKey.kms = this.assertedKms(kms)
+    minimalImportableKey.type = this.assertedKeyType({ key, type })
+    minimalImportableKey.meta = { purposes: this.assertedPurposes({ key, type }) }
+    return minimalImportableKey as MinimalImportableKey
+  }
+
   private assertedKms(kms?: string) {
     const result = kms ?? this.defaultKms
     if (!!result) {
       return result
     }
     throw Error('no KMS supplied')
   }
+
+  private assertedKeyType = (args: { key?: IKeyOpts; type: EbsiKeyType }): EbsiKeyType => {
+    if (!args.key?.type) {
+      return args.type
```

Comment:
Actually this method asserts nothing. It just picks the type from a specific location

Reply:
Renamed to `setDefaultKeyType`